.github/workflows/main.yml

```diff
@@ -0,0 +1,267 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - '**'
+  create:
+    tags:
+      - '**'
+
+jobs:
+  linux-basic:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'create'
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install libxss1 libgstreamer-plugins-base1.0-0 libgstreamer1.0-0 qt5-default libqt5gui5 libqt5webengine5 libqt5webenginecore5 libqt5webenginewidgets5 libqt5printsupport5 libqt5quickwidgets5 libqt5x11extras5 libxss1 
+        sudo apt install patchelf binutils cmake pkg-config qtbase5-dev qtwebengine5-dev libqt5x11extras5-dev  qtbase5-private-dev libssl-dev libxss-dev libxmu-dev
+    - name: Build the Linux binary
+      run: |
+        bash ./dist/linux/package.sh build
+    - name: Package the binaries, create tarball
+      if: github.event_name == 'create'
+      run: |
+        bash ./dist/linux/package.sh compose
+    - name: Upload to GitHub
+      if: github.event_name == 'create'
+      run: |
+        EVENT_TYPE=$(cat "$GITHUB_EVENT_PATH" | jq -r ".ref_type")
+        if [[ "$EVENT_TYPE" != "tag" ]]; then exit 0; fi
+        TAG_NAME=$(cat "$GITHUB_EVENT_PATH" | jq -r ".ref")
+        ./dist/upload-github-release-asset.sh github_api_token=${{ secrets.GITHUB_TOKEN }} owner=toggl repo=toggldesktop tag="$TAG_NAME" filename="toggldesktop_x86_64.tar.gz"
+
+  linux-tarball:
+    runs-on: ubuntu-16.04
+    if: github.event_name == 'create'
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install libxss1 libgstreamer-plugins-base1.0-0 libgstreamer1.0-0
+        sudo apt install patchelf binutils cmake pkg-config libssl-dev libxss-dev libxmu-dev
+        sudo apt install dbus libfreetype6 fontconfig libx11-6 libx11-xcb1 libgl1-mesa-dev libnss3 libasound2 libxcomposite1 libxcursor1 libxi6 libxtst6 wget
+    - name: Fetch and install binary Qt
+      run: |
+        bash ./dist/linux/install-qt.sh 5.12.5
+    - name: Build the Linux binary
+      run: |
+        export CMAKE_PREFIX_PATH="${PWD}/Qt/5.12.5/gcc_64/lib/cmake/"
+        bash ./dist/linux/package.sh build
+    - name: Package the binaries, create tarball
+      run: |
+        export CMAKE_PREFIX_PATH="${PWD}/Qt/5.12.5/gcc_64/lib/cmake/"
+        bash ./dist/linux/package.sh compose
+    - name: Upload to GitHub
+      run: |
+        EVENT_TYPE=$(cat "$GITHUB_EVENT_PATH" | jq -r ".ref_type")
+        if [[ "$EVENT_TYPE" != "tag" ]]; then exit 0; fi
+        TAG_NAME=$(cat "$GITHUB_EVENT_PATH" | jq -r ".ref")
+        ./dist/upload-github-release-asset.sh github_api_token=${{ secrets.GITHUB_TOKEN }} owner=toggl repo=toggldesktop tag="$TAG_NAME" filename="toggldesktop_x86_64.tar.gz" renameto="toggldesktop_linux_${TAG_NAME/v/}_x86_64.tar.gz"
+
+  linux-ubuntu1804:
```

Comment:
there's actually a second branch (or is that in my private repo? dunno, but i'm going to open another PR to add that to the flow when this is merged) called linux-ubuntu1604 because we'll need to build a special package for older ubuntu
i initially did the deb package with newer packages which means it is only 3MB large but it doesn't run on ubuntu 16.04
meanwhile the ubuntu 16.04 package will support a broader range of distros but it's ~ 80MB large